src/stubgen.py

```diff
@@ -81,11 +82,63 @@ class and repeatedly call ``.put()`` to register modules or contents within the
 ]
 # fmt: on
 
+# (name, desired_as_name) -> actual_as_name
+ImportDict = Dict[Tuple[Optional[str], Optional[str]], Optional[str]]
+# package_name -> ((name, desired_as_name) -> actual_as_name)
+PackagesDict = Dict[str, ImportDict]
+
+# (signature_str, doc_str, default_arg_1, default_arg_2, ...)
+# We could type as Tuple[Optional[str], Optional[str], *Tuple[Any, ...]]
+# but this is only valid in python >= 3.11
+NbSignature = Tuple[Optional[str], Optional[str], Any]
+NbGetterSetterSignature = Tuple[str, str]
+
+
+class NbObject(Protocol):
```

Comment:
I think that these protocol all need a 1-line comment explaining what they could be. For example, this one is called `NbObject`, but I'm not sure if all nanobind-created objects have a ``_nb_signature__``. Type objects do not, and neither does the metaclass.

Reply:
See also this.